components/Tabs/src/TabBarView/MDCTabBarView.m

```diff
@@ -420,100 +406,109 @@ - (void)observeValueForKeyPath:(NSString *)keyPath
 - (void)layoutSubviews {
   [super layoutSubviews];
 
-  CGRect availableBounds = self.bounds;
-  if (@available(iOS 11.0, *)) {
-    availableBounds = UIEdgeInsetsInsetRect(availableBounds, self.safeAreaInsets);
-  }
-  CGFloat availableWidth = CGRectGetWidth(availableBounds);
-  CGFloat requiredWidth = [self justifiedWidth];
-  BOOL canBeJustified = availableWidth >= requiredWidth;
-  if (canBeJustified) {
-    [NSLayoutConstraint deactivateConstraints:self.scrollableLayoutConstraints];
-    self.containerView.distribution = UIStackViewDistributionFillEqually;
-    [NSLayoutConstraint activateConstraints:self.justifiedLayoutConstraints];
+  if (self.isJustifiedLayoutStyle) {
+    [self layoutSubviewsForJustifiedLayout];
   } else {
-    [NSLayoutConstraint deactivateConstraints:self.justifiedLayoutConstraints];
-    self.containerView.distribution = UIStackViewDistributionFillProportionally;
-    [NSLayoutConstraint activateConstraints:self.scrollableLayoutConstraints];
+    [self layoutSubviewsForScrollableLayout];
   }
+  self.contentSize = [self calculatedContentSize];
 
   if (!self.initialScrollDone) {
     self.initialScrollDone = YES;
     [self scrollUntilSelectedItemIsVisibleWithoutAnimation];
   }
 }
 
-- (void)willMoveToSuperview:(UIView *)newSuperview {
-  [super willMoveToSuperview:newSuperview];
-  self.initialScrollDone = NO;
+- (BOOL)isJustifiedLayoutStyle {
+  CGRect availableBounds = self.bounds;
```

Comment:
Tip: Refactor this out to a shared method

```objc
- (CGRect)availableBounds {
  CGRect availableBounds = self.bounds;
  if (@available(iOS 11.0, *)) {
    availableBounds = UIEdgeInsetsInsetRect(availableBounds, self.adjustedContentInset);
  }
  return availableBounds;
}
```

Reply:
+1

Reply:
Done.